collectors/scm/github/src/main/java/com/capitalone/dashboard/collector/GitHubCollectorTask.java

```diff
@@ -166,9 +166,28 @@ public void collect(Collector collector) {
         int commitCount = 0;
         int pullCount = 0;
         int issueCount = 0;
-
+       
         clean(collector);
-        for (GitHubRepo repo : enabledRepos(collector)) {
+        
+        String proxyUrl = gitHubSettings.getProxy();
+        String proxyPort = gitHubSettings.getProxyPort();
+        String proxyUser= gitHubSettings.getProxyUser();
+        String proxyPassword= gitHubSettings.getProxyPassword();
+        
+        if (!StringUtils.isEmpty(proxyUrl) && !StringUtils.isEmpty(proxyPort)) {
+            System.setProperty("http.proxyHost", proxyUrl);
+            System.setProperty("https.proxyHost", proxyUrl);
+            System.setProperty("http.proxyPort", proxyPort);
+            System.setProperty("https.proxyPort", proxyPort);
+        }
+         if (!StringUtils.isEmpty(proxyUser) && !StringUtils.isEmpty(proxyPassword)) {
```

Comment:
This if block would need to be in the above if block

Reply:
i don't think that change is required. that if (183) inside other if statement make sense. because username and password is valid only if proxy url is given . without url and port no .user id and password make no sense

Reply:
I’d prefer for this to be inside the if block on 177. This is because if the Url and port doesn’t exist why even check to see if the user and password exist?